src/modules/cmdpal/Microsoft.CmdPal.UI/MainWindow.xaml.cs

```diff
@@ -232,6 +233,16 @@ private void ShowHwnd(IntPtr hwndValue, MonitorBehavior target)
         PositionCentered(display);
 
         PInvoke.ShowWindow(hwnd, SHOW_WINDOW_CMD.SW_SHOW);
+
+        // instead of showing the window, uncloak it from DWM
+        // This will make it visible to the user, without the animation or frames for
+        // loading XAML with composition
+        unsafe
+        {
+            BOOL value = false;
+            PInvoke.DwmSetWindowAttribute(_hwnd, DWMWINDOWATTRIBUTE.DWMWA_CLOAK, (void*)&value, (uint)sizeof(BOOL));
```

Comment:
does this mean that window scanning tools will erroneously consider it to be visible? since DWM isn't part of the user32 window model

Reply:
IMO "some apps that report the visible windows but don't know how to deal with CLOAK'd hwnds are wrong for a few releases" is a less bad problem than "CmdPal flash-bangs all of our users"